src/persist-client/src/read.rs

```diff
@@ -252,10 +380,18 @@ where
         as_of: Antichain<T>,
     ) -> Result<Result<Listen<K, V, T, D>, InvalidUsage>, LocationError> {
         trace!("ReadHandle::listen timeout={:?} as_of={:?}", timeout, as_of);
+        if PartialOrder::less_than(&as_of, &self.since) {
```

Comment:
this impl feels kinda sus given that it doesn't need to call into the state machine (ReadHandle::snapshot calls a read-only method on Machine)